gobbler-server.rkt

```diff
@@ -232,17 +235,61 @@ waiting list.
 
 ;; GobblerUniverse iworld? sexp? -> GobblerBundle
 ;; Update the waypoint of the player
-(define (update-waypoint uni world sexp)
-  uni)
+(define (update-waypoint! uni world sexp)
+  (cond [(playing? uni)
+         (cond [(waypoint-message? sexp)
+                (handle-waypoint! uni world sexp)]
+               [else uni])]
+        [else uni]))
+
+;; Playing iworld? WaypointMessage -> GobblerBundle
+;; Update the waypoint of the player
+(define (handle-waypoint! playing world waypoint-msg)
+  (define waypoint (waypoint-message->posn waypoint-msg))
+  (define player (get-player playing world))
 
+  (when (and (waypoint-reachable? waypoint) player)
+    (set-turkey-waypoint! (player-turkey player) waypoint))
+  playing)
 
 ;; =====================================
 ;; UTILS
 ;; =====================================
 
-;; two helper functions that rely on domain knowledge from geometry
+;; Playing iworld? -> [Maybe Player]
+;; Returns the player that corresponds with the given iworld or false if no such
+;; player exists
+(define (get-player playing world)
+  (findf (λ (p) (iworld=? world (player-iworld p))) (ready-players playing)))
+
+;; sexp? -> Boolean
+;; Determines if the given sexp is a valid well-formed message
+(define (c2s-message? sexp)
+  (waypoint-message? sexp))
+
+;; sexp? -> Boolean
+;; Determines if the given sexp is a well-formed waypoint message
+(define (waypoint-message? sexp)
+  (and (list? sexp)
+       (= (length sexp) 3)
+       (symbol? (first sexp))
+       (symbol=? (first sexp) WAYPOINT)
+       (number? (second sexp))
+       (number? (third sexp))))
+
+;; WaypointMessage -> Posn
+;; Converts the given waypoint message into a posn
+(define (waypoint-message->posn waypoint)
+  (posn (second waypoint) (third waypoint)))
+
+;; Posn -> Boolean
+;; Determines if the waypoint is valid
+(define (waypoint-reachable? posn)
+  (and (>= (posn-x posn) 0)
+       (>= (posn-y posn) 0)
+       (<= (posn-x posn) GAME-SIZE)
+       (<= (posn-y posn) GAME-SIZE)))
```

Comment:
I would abstract this out to an `in-range?` function that takes the value, the min, and the max.

Reply:
Wouldn't that then have to take a max for x and y? thats five things to pass in, I'd rather leave this like it is and then do a posn in range function later if we need it. I can rewrite the comparisons to be cleaner though, I forgot I could do that until just now

Reply:
I meant that you would call this `in-range?` function for each both the x and y dimensons. i.e.
```racket
(define (in-range? val min max)
  (and (>= val min) (<= val max)))
```

And then this section becomes:
```racket
(and (in-range? (posn-x posn) 0 GAME-SIZE)
     (in-range? (posn-y posn) 0 GAME-SIZE))
```

You could even go one step further and make it `in-game-range?` and hard-code the `0` and `GAME-SIZE` bounds.

Reply:
To me, that properly abstracts the similarities expressed here.